File: app/src/main/java/life/plank/juna/zone/view/LatestMatch/LeagueBinder.java
package life.plank.juna.zone.view.LatestMatch;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ahamed.multiviewadapter.BaseViewHolder;
import com.ahamed.multiviewadapter.ItemBinder;

import life.plank.juna.zone.R;

class LeagueBinder extends ItemBinder<LeagueModel, LeagueBinder.LeagueViewHolder> {

    @Override
    public LeagueViewHolder create(LayoutInflater inflater, ViewGroup parent) {
        return new LeagueViewHolder(inflater.inflate(R.layout.item_leagues, parent, false));
    }

    @Override
    public void bind(LeagueViewHolder holder, LeagueModel item) {
        // Bind the data here
    }

    @Override
    public boolean canBindData(Object item) {
        return item instanceof LeagueModel;
    }

    class LeagueViewHolder extends BaseViewHolder<LeagueModel> {

        LeagueViewHolder(View itemView) {
            super(itemView);
        }
        // Normal ViewHolder code
    }
}

File: app/src/main/java/life/plank/juna/zone/view/latestMatch/LeagueBinder.kt
package life.plank.juna.zone.view.latestMatch

import android.app.Activity
import android.view.*
import com.ahamed.multiviewadapter.*
import com.prembros.facilis.util.onReducingClick
import kotlinx.android.synthetic.main.item_league.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.interfaces.OnItemClickListener
import life.plank.juna.zone.util.common.JunaDataUtil.*
import life.plank.juna.zone.view.activity.base.BaseJunaCardActivity
import life.plank.juna.zone.view.fragment.football.LeagueInfoFragment

class LeagueBinder(private val onItemClickListener: OnItemClickListener, private val activity: Activity) : ItemBinder<LeagueModel, LeagueBinder.LeagueViewHolder>() {

    override fun create(inflater: LayoutInflater, parent: ViewGroup): LeagueViewHolder = LeagueViewHolder(inflater.inflate(R.layout.item_league, parent, false))

    override fun bind(holder: LeagueViewHolder, item: LeagueModel) {
        holder.itemView.all_leagues_card.onReducingClick(0) {
            onItemClickListener.onItemClicked()
        }

        holder.itemView.epl_card.onReducingClick {
            (activity as? BaseJunaCardActivity)?.pushFragment(LeagueInfoFragment.newInstance(
                    getSpecifiedLeague(findString(R.string.premier_league))), true)
        }

        holder.itemView.efl_card.onReducingClick {
            (activity as? BaseJunaCardActivity)?.pushFragment(LeagueInfoFragment.newInstance(
                    getSpecifiedLeague(findString(R.string.champions_league))), true)
        }
    }

    override fun canBindData(item: Any): Boolean = item is LeagueModel

    class LeagueViewHolder(itemView: View) : ItemViewHolder<LeagueModel>(itemView)
}